Remove GCE autoscaling metrics from GCM sink

bskiba opened the pull request:
Fix #1487. Remove GCE autoscaling metrics from GCM sink. GCE Autoscaler has not
been supported in Kubernetes for a while. Also remove the `metrics`
parameter of GCM sink configuration, which was previously used to enable
collection of only GCE autoscaling metrics.

bskiba committed:
Remove GCE autoscaling metrics from GCM sink

Remove GCE autoscaling metrics from GCM sink. GCE Autoscaler has not
been supported in Kubernetes for a while. Also remove the `metrics`
parameter of GCM sink configuration, which was previously used to enable
collection of only GCE autoscaling metrics.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/heapster/1665)
<!-- Reviewable:end -->

bskiba commented:
Bump to verify cla.

piosz commented on metrics/core/metrics.go:
```diff
@@ -73,17 +73,6 @@ var LabeledMetrics = []Metric{
 }
 
-var NodeAutoscalingMetrics = []Metric{
-	MetricNodeCpuCapacity,
```
Let's leave metrics:
```
MetricNodeCpuCapacity,
MetricNodeMemoryCapacity,		
MetricNodeCpuAllocatable,		
MetricNodeMemoryAllocatable,
```
which are generally useful.

Let's also rename `NodeAutoscalingEnricher` to just `NodeEnricher`

piosz commented:
@mwielgus

bskiba committed:
Keep useful metrics

Revert deletion of following metrics:
MetricNodeCpuCapacity,
MetricNodeMemoryCapacity,
MetricNodeCpuAllocatable,
MetricNodeMemoryAllocatable,
Rename NodeAutoscalingEnricher to NodeEnricher

bskiba committed:
Fix merge issues

piosz commented:
Closing for now as obsolete.
